### ISSUE TITLE
 some differences from Ghorbani‘s code

### ISSUE THREAD

**Issue by splendiddiddid** [maxtli/neuronshapley#1]
Hi, I was very surprised to find this PyTorch-based code repository! However, while reading the code, I found several points that differed from the original code repository of Ghorbani et al.

- the funtion of `update_once()` in mab.py which used to update the neuron shapley value in one permutation :

1. in line 154  you  wrote `differential = loss - prev_loss`  , I feel very confused, why not  `differential = prev_loss - loss` ? 
2. in line 158 you wrote `value_update = (differential - shapley_values[neuron]) / samples[neuron]`, I think maybe this `value-update`is the **_marginal contribute_** of this neuron，but I also feel very confused,  why is it divided by  `samples[neuron]` and why did it need to subtract `  shapley_values[neuron] `  here?    because in Ghorbani‘s repository ，there just is `  marginals[c[idx]] = (old_val - new_val) / len(c[idx])`  in cb_run.py in line 91；

- the function of ` compute_relevant_neurons()`  in mab.py which used to update the candidate neurons：

1. in line 103 you wrote  `threshold = np.partition(shapley_values, -k)[-k]`, but in Ghorbani‘s repository ，there  is `  thresh = (vals)[np.argsort(vals)[-top_k - 1]]`  in cb_aggregate.py in line 99. I'm really confused about whether it makes a difference here if we use k or k-1;
2. in line 111 and 112, your code is ` if shapley_values[i] + cb[i] > threshold + epsilon and shapley_values[i] - cb[i] < threshold - epsilon`, but in Ghorbani‘s repository ，there  is `chosen_players = np.where( ((vals - cbs) < thresh) * ((vals + cbs) > thresh))[0]`  in cb_aggregate.py in line 100, they didn't use the  variable epsilon.

Because I was not familiar with the tensorflow framework , I could not read the Ghorbani‘s code, so I was very surprised to find this PyTorch-based code repository. But I noticed some differences in your core code, and I was confused. I'm sorry to bother you. Looking forward to your reply
